unstructured/file_utils/filetype.py

```diff
@@ -105,102 +105,183 @@ def __lt__(self, other):
+            return FileType.JPG
 
-    elif mime_type == "image/png":
-        return FileType.PNG
+        elif mime_type == "image/png":
+            return FileType.PNG
 
-    elif mime_type == "text/plain":
-        if extension and extension == ".eml":
-            return FileType.EML
-        if file and not extension:
-            if _check_eml_from_buffer(file=file) is True:
+        elif mime_type == "text/plain":
+            if extension and extension == ".eml":
                 return FileType.EML
+            if file and not extension:
+                if _check_eml_from_buffer(file=file) is True:
+                    return FileType.EML
+                else:
+                    return FileType.TXT
             else:
                 return FileType.TXT
-        else:
-            return FileType.TXT
 
-    elif mime_type.endswith("xml"):
-        if extension and extension == ".html":
+        elif mime_type.endswith("xml"):
+            if extension and extension == ".html":
+                return FileType.HTML
+            else:
+                return FileType.XML
+
+        elif mime_type == "text/html":
             return FileType.HTML
-        else:
-            return FileType.XML
 
-    elif mime_type == "text/html":
-        return FileType.HTML
+        elif mime_type.startswith("text"):
+            return FileType.TXT
 
-    elif mime_type.startswith("text"):
-        return FileType.TXT
+        elif mime_type in XLSX_MIME_TYPES:
+            return FileType.XLSX
 
-    elif mime_type in XLSX_MIME_TYPES:
-        return FileType.XLSX
+        elif mime_type in XLS_MIME_TYPES:
+            return FileType.XLS
 
-    elif mime_type in XLS_MIME_TYPES:
-        return FileType.XLS
+        elif mime_type in PPTX_MIME_TYPES:
+            return FileType.PPTX
 
-    elif mime_type in PPTX_MIME_TYPES:
-        return FileType.PPTX
+        elif mime_type in PPT_MIME_TYPES:
+            return FileType.PPT
 
-    elif mime_type in PPT_MIME_TYPES:
-        return FileType.PPT
+        elif mime_type == "application/octet-stream":
+            if file and not extension:
+                return _detect_filetype_from_octet_stream(file=file)
+            else:
+                return EXT_TO_FILETYPE.get(extension, FileType.UNK)
+
+        elif mime_type == "application/zip":
+            filetype = FileType.UNK
+            if file and not extension:
+                filetype = _detect_filetype_from_octet_stream(file=file)
+            elif filename is not None:
+                with open(filename, "rb") as f:
+                    filetype = _detect_filetype_from_octet_stream(file=f)
+
+            if filetype == FileType.UNK:
+                return FileType.ZIP
+            else:
+                return filetype
+
+        logger.warn(
+            f"MIME type was {mime_type}. This file type is not currently supported in unstructured."
+        )
+        return FileType.UNK
+    except:
```

Comment:
We'll want to make this reference the specific exception instead of having it as a bare except. Should be `except ImportError:`. This is what I get on `import magic` when `libmagic` isn't available:

```python
ImportError: failed to find libmagic.  Check your installation
```